### ISSUE TITLE
Customer Extended

### ISSUE THREAD

**Issue by ambs02** [UQcsse3200/2024-studio-3#101]
# **Description**
Built on top of the previously closed Customers feature ticket, this feature ticket will build the base for full customer interaction and behaviour. As a “level” starts, the game will generate customers that will spawn using their correct sprite (animation is a stretch goal). These customers will move and wait by the counter, and the player will be able to interact with them, taking their order and serving them their food. Customer movement will be fully implemented, once the customers patience/wait time has run out, the customer will become unhappy and leave the restaurant. 

## SubFeature/component (optional)
- spawning personality customers with the correct sprite
- customer dialogue for ordering
- integrating customer ordering with docket/ticket system
- customers leave after a certain amount of time and the player’s reputation with them lowers

# **Dependencies**
Feature ticket relies on Customers #7, Order System #8, Map Design #1

# **Milestones**
- [x] #159
- [x] #136
- [x] #175
- [x] #221

**Completion Deadline:** Sept. 13

# **Documentation**
- [Customers](https://github.com/UQcsse3200/2024-studio-3/wiki/Customers)
- [Ordering System](https://github.com/UQcsse3200/2024-studio-3/wiki/Ordering-System-Overview)
- [Map Design](https://github.com/UQcsse3200/2024-studio-3/wiki/Map-Design)

# **Members**
Aiden Richards (@AidenRichards03)
Marcus Baulch (@baulch-m)
Vansh Mittal (@V4N5HM)
Lei Su (@ImLeiSu)
